### PR TITLE
load data: fix bug if load data with long content

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29240

Problem Summary:

### Check List

Tests 
- [ ] Manual test (add detailed scripts or steps below)

load the file manually

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — COMMENTED** [review]
How about adding a unit test for this function?

**Comment by bb7133** [comment]
> How about adding a unit test for this function?

Please add a test case for your change.

File: executor/load_data.go [hunk 1]
```diff
@@ -459,13 +459,10 @@ func (e *LoadDataInfo) indexOfTerminator(bs []byte, isInQuoter bool) int {
...
 	}
 	atFieldStart := true
-	inQuoter := false
 loop:
 	for i := 0; i < len(bs); i++ {
 		if atFieldStart && bs[i] == e.FieldsInfo.Enclosed {
-			if !isInQuoter {
-				inQuoter = true
-			}
+			inQuoter = !inQuoter
```
Comment (djshow832) [inline executor/load_data.go:465]: So all cases where one string crosses multiple lines will cause an error, right?
Is it possible to verify that in the test case?

Comment (xiongjiwei) [reply executor/load_data.go:465]: not all cases, this error will only happen if one string is split into 2 parts and crosser multiple lines. e.g. first part `"123` second part `456\n789"` the `\n` in the second part should not interpret as new lines.

the test case  No.4 test this scenario

**Review by wjhuang2016 — APPROVED** [review]

**Review by djshow832 — APPROVED** [review]

**Pull request merged**